### PR TITLE
feat(console,phrases): add application permissions edit modal

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add the application permissions edit modal.

Resource scopes and organization scope descriptions can be directly editable through the third-party permissions management page. 

image

## Testing

test locally

## Checklist

- [ ] ~`.changeset`~
- [ ] ~unit tests~
- [ ] ~integration tests~
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/console/src/pages/ApplicationDetails/components/Permissions/ApplicationScopesManagementModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  const tAction = useActionTranslation();
+
+  const onSubmitHandler = handleSubmit(trySubmitSafe(onSubmit));
+
+  // Reset form on open
+  useEffect(() => {
+    if (scope) {
+      reset(scope);
+    }
+  }, [reset, scope]);
```
Comment (xiaoyijun) [inline packages/console/src/pages/ApplicationDetails/components/Permissions/ApplicationScopesManagementModal/index.tsx:52]: How about use the `scope` as the default value of the `useForm` hook? then we can remove this useEffect.

Comment (simeng-li) [reply packages/console/src/pages/ApplicationDetails/components/Permissions/ApplicationScopesManagementModal/index.tsx]: default value won't update with the input props.

File: packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Permissions/index.tsx [hunk 2]
```diff
@@ -22,7 +29,9 @@ function Permissions({ application }: Props) {
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
 
   const [isAssignScopesModalOpen, setIsAssignScopesModalOpen] = useState(false);
-  const { parseRowGroup, deleteScope } = useScopesTable();
+  const [editScopeModalData, setEditScopeModalData] = useState<EditableScopeData | undefined>();
```
Comment (charIeszhao) [inline packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Permissions/index.tsx:32]: nit: `| undefined` can be saved

**Review by charIeszhao — APPROVED** [review]

**Review by xiaoyijun — APPROVED** [review]
LGTM

File: packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Permissions/index.tsx (hunk 2 above)
Comment (simeng-li) [reply packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Permissions/index.tsx]: updated

**Pull request merged**